components/dashboard/src/settings/Integrations.tsx

```diff
@@ -199,77 +179,435 @@ function GitProviders() {
+            </div>
+            {props.mode === "new" && (
+                <div className="flex flex-col space-y-2">
+                    <label htmlFor="type" className="font-medium">Provider Type</label>
+                    <select name="type" value={type} disabled={props.mode !== "new"} className="rounded-md w-full"
+                        onChange={(e) => setType(e.target.value)}>
+                        <option value="GitHub">GitHub</option>
+                        <option value="GitLab">GitLab</option>
+                    </select>
+                </div>
+            )}
+            <div className="flex flex-col space-y-2">
+                <label htmlFor="hostName" className="font-medium">Provider Host Name</label>
+                <input name="hostName" disabled={props.mode === "edit"} type="text" value={host} className="rounded-md w-full"
+                    onChange={(e) => updateHostValue(e.target.value)} />
+            </div>
+            <div className="flex flex-col space-y-2">
+                <label htmlFor="redirectURL" className="font-medium">Redirect URL</label>
+                <div className="w-full relative">
+                    <input name="redirectURL" disabled={true} readOnly={true} type="text" value={redirectURL} className="rounded-md w-full" />
+                    <div className="cursor-pointer" onClick={() => copyRedirectUrl()}>
+                        <img src={images.copy} className="absolute top-1/3 right-3" />
+                    </div>
+                </div>
+                <span className="text-gray-500 text-sm">{getRedirectUrlDescription(type, host)}</span>
+            </div>
+            <div className="flex flex-col space-y-2">
+                <label htmlFor="clientId" className="font-medium">Client ID</label>
+                <input name="clientId" type="text" value={clientId} className="rounded-md w-full"
+                    onChange={(e) => updateClientId(e.target.value)} />
+            </div>
+            <div className="flex flex-col space-y-2">
+                <label htmlFor="clientSecret" className="font-medium">Client Secret</label>
+                <input name="clientSecret" type="password" value={clientSecret} className="rounded-md w-full"
+                    onChange={(e) => updateClientSecret(e.target.value)} />
+            </div>
+            {errorMessage && (
+                <div className="flex rounded-md bg-red-600 p-3">
+                    <span className="text-white">{errorMessage}</span>
+                </div>
+            )}
+            {validationError && (
+                <div className="flex rounded-md bg-red-600 p-3">
+                    <span className="text-white">{validationError}</span>
+                </div>
+            )}
+        </div>
+        <div className="flex justify-end mt-6">
+            <button className="disabled:opacity-50" onClick={() => validate() && activate()} disabled={!!validationError || busy}>Activate Integration</button>
```

Comment:
**question**: Could we also disable the button on first load when someone is trying to add an integration?

Reply:
sounds reasonable!